stripe-core/src/main/java/com/stripe/android/core/networking/DefaultAnalyticsRequestExecutor.kt

```diff
@@ -11,45 +21,137 @@ import kotlin.coroutines.CoroutineContext
     ) : this(
-        DefaultStripeNetworkClient(
+        context = context,
+        stripeNetworkClient = DefaultStripeNetworkClient(
+            workContext = Dispatchers.IO,
+            logger = Logger.getInstance(BuildConfig.DEBUG),
+        ),
+        workContext = Dispatchers.IO,
+    )
+
+    constructor(
+        context: Context,
+        workContext: CoroutineContext,
+    ) : this(
+        context = context,
+        stripeNetworkClient = DefaultStripeNetworkClient(
             workContext = workContext,
-            logger = logger
+            logger = Logger.getInstance(BuildConfig.DEBUG),
         ),
-        workContext,
-        logger
+        workContext = workContext,
     )
 
     /**
      * Make the request and ignore the response.
      */
     override fun executeAsync(request: AnalyticsRequest) {
-        logger.info("Event: ${request.params[FIELD_EVENT]}")
+        if (canUseWorkManager) {
+            val workManager = WorkManager.getInstance(context)
+            enqueue(workManager, request)
+        } else {
+            execute(request)
+        }
+    }
+
+    private fun execute(request: AnalyticsRequest) {
+        val logger = Logger.getInstance(enableLogging = BuildConfig.DEBUG)
+        val eventName = request.params[FIELD_EVENT]
+        logger.info("Event: $eventName")
 
         CoroutineScope(workContext).launch {
             runCatching {
                 stripeNetworkClient.executeRequest(request)
+            }.onSuccess {
+                logger.info("Successfully sent event $eventName")
             }.onFailure {
-                logger.error("Exception while making analytics request", it)
+                logger.error("Exception while sending event $eventName", it)
             }
         }
     }
 
+    /**
+     * Make the request and ignore the response.
+     */
+    private fun enqueue(
+        workManager: WorkManager,
+        request: AnalyticsRequest
+    ) {
+        val inputData = Data.Builder()
+            .putSerializable(FIELD_DATA, request)
+            .build()
+
+        val constraints = Constraints.Builder()
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .setRequiresBatteryNotLow(true)
+            .build()
+
+        val workRequest = OneTimeWorkRequestBuilder<SendAnalyticsEventWorker>()
+            .addTag("AnalyticsRequest")
+            .setInputData(inputData)
+            .setConstraints(constraints)
+            .build()
+
+        workManager.enqueue(workRequest)
+    }
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    class SendAnalyticsEventWorker(
+        appContext: Context,
+        params: WorkerParameters,
+    ) : CoroutineWorker(appContext, params) {
+
+        override suspend fun doWork(): Result {
+            val request = inputData.getSerializable<AnalyticsRequest>(FIELD_DATA)
+                ?: return Result.failure()
+
+            val stripeNetworkClient = WorkManagerHelpers.getOrCreateNetworkClient()
```

Comment:
We can’t inject anything into the `Worker` as it will be created via reflection, so using this to create and store a network client.

Reply:
sounds good! different SDKs might require different network client implementations for their analytics in the future, if so we can figure how to provide a custom network client builder